File: scripts/db/V001_CreateCartsCollection.js
// ============================================================
// Emart Cart DB - Standalone Migration Scripts
// DB Team: Run these directly with mongosh if needed outside
// the Go Mongock runner (e.g. CI/CD pre-check or emergency)
//
// Usage:
//   mongosh "mongodb://user:pass@host:27017" --file V001_CreateCartsCollection.js
//
// IMPORTANT: The Go cart service runs these automatically on
// startup via the Mongock-style migration runner. Only run
// these manually if directed by DB lead.
// ============================================================

// ============================================================
// V001 - Create carts collection
// ============================================================
use('cart');

print('=== V001_CreateCartsCollection START ===');

var collections = db.getCollectionNames();
if (!collections.includes('carts')) {
    db.createCollection('carts');
    print('  ✓ carts collection created');
} else {
    print('  ⏭ carts collection already exists - skipping');
}

// Record in Mongock change log (mirrors what Go runner does)
db.mongockChangeLog.updateOne(
    { changeId: 'V001_CreateCartsCollection' },
    {
        $set: {
            changeId:   'V001_CreateCartsCollection',
            author:     'emart-db-team',
            state:      'EXECUTED',
            executedAt: new Date(),
            order:      '001',
            source:     'manual-script'
        }
    },
    { upsert: true }
);

print('=== V001_CreateCartsCollection DONE ===');

File: cart-service/internal/migration/v001_create_carts_collection.go
package migration

import (
	"context"
	"go.mongodb.org/mongo-driver/mongo"
)

// V001CreateCartsCollection creates the 'carts' collection in the 'cart' database
type V001CreateCartsCollection struct{}
func NewV001CreateCartsCollection() *V001CreateCartsCollection { return &V001CreateCartsCollection{} }
func (m *V001CreateCartsCollection) ID() string     { return "V001_CreateCartsCollection" }
func (m *V001CreateCartsCollection) Order() string  { return "001" }
func (m *V001CreateCartsCollection) Author() string { return "emart-db-team" }

func (m *V001CreateCartsCollection) Execute(ctx context.Context, db *mongo.Database) error {
	// Create carts collection
	names, _ := db.ListCollectionNames(ctx, map[string]interface{}{})
	exists := false
	for _, n := range names { if n == "carts" { exists = true; break } }
	if !exists {
		return db.CreateCollection(ctx, "carts")
	}
	return nil
}

func (m *V001CreateCartsCollection) Rollback(ctx context.Context, db *mongo.Database) error {
	return db.Collection("carts").Drop(ctx)
}
